packages/components/src/Chips/InternalChipDismissible/__tests__/KeyboardNavigation.test.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useState } from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { InternalChipDismissible } from "..";
+import { Chip } from "../..";
+
+// Helper function to get a chip by label
+function getByChipLabelText(chipName: string) {
+  return screen.getByLabelText(
+    chipName + ". Press delete or backspace to remove " + chipName,
+  );
+}
+
+// Helper to get remove button within a chip
+function getRemoveButton(chip: HTMLElement) {
+  const button = chip.querySelector('[data-testid="remove-chip-button"]');
+
+  if (!button) {
+    throw new Error("Remove button not found");
```

Comment:
using a proper testing-library selector saves us from having to write anything like this. use that instead if possible.